src/librustdoc/passes/html_tags.rs

```diff
@@ -142,39 +142,30 @@ fn extract_html_tag(
     }
 }
 
-fn extract_html_comment(
-    text: &str,
-    range: &Range<usize>,
-    start_pos: usize,
-    iter: &mut Peekable<CharIndices<'_>>,
-    f: &impl Fn(&str, &Range<usize>),
-) {
-    // We first skip the "!--" part.
-    let mut iter = iter.skip(3);
-    while let Some((pos, c)) = iter.next() {
-        if c == '-' && text[pos..].starts_with("-->") {
-            // All good, we can leave!
-            return;
-        }
-    }
-    f(
-        "Unclosed HTML comment",
-        &Range { start: range.start + start_pos, end: range.start + start_pos + 3 },
-    );
-}
-
 fn extract_tags(
     tags: &mut Vec<(String, Range<usize>)>,
     text: &str,
     range: Range<usize>,
+    is_in_comment: &mut Option<Range<usize>>,
     f: &impl Fn(&str, &Range<usize>),
 ) {
     let mut iter = text.char_indices().peekable();
 
     while let Some((start_pos, c)) = iter.next() {
-        if c == '<' {
+        if is_in_comment.is_some() {
+            if c == '-' && text[start_pos..].starts_with("-->") {
+                *is_in_comment = None;
+            }
+        } else if c == '<' {
             if text[start_pos..].starts_with("<!--") {
-                extract_html_comment(text, &range, start_pos, &mut iter, f);
+                // We skip the "!--" part. (Once `advance_by` is stable, might be nice to use it!)
```

Comment:
Rustdoc uses nightly, why not use it now?

Reply:
I'm not sure if the API will remain the same. Currently it looks awkward, it forces you to check what the function returns (because it returns a `Result`) but it's completely unnecessary in my opinion because an `Err` isn't an error in itself...

Reply:
Can you mention that on the tracking issue? https://github.com/rust-lang/rust/issues/77404 This sort of thing is exactly why we use dogfooding, to get more feedback on the APIs :)

Reply:
Done!